### PR TITLE
[SPARK-49208][SQL][TESTS] Add more tests for negative MONTH intervals

### PR DESCRIPTION
**Pull request by nemanjapetr-db**
### What changes were proposed in this pull request?
Adding more tests.

### Why are the changes needed?
To cover particular values of negative months that might be susceptible to bugs due to modulo operations on negative numbers.

### Does this PR introduce _any_ user-facing change?
No. Test only.

### How was this patch tested?
Ran the tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by MaxGekk — COMMENTED** [review]
The changes you pointed out in SPARK-35680 were released long time ago. Don't think your changes could be considered as a follow up. Please, remove the tag `[FOLLOWUP]` and add `[SQL]` instead.

**Comment by MaxGekk** [comment]
@nemanjapetr-db Please, open a separate JIRA for the changes, and set it to PR's title.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala [hunk 1]
```diff
@@ -670,6 +670,22 @@ class IntervalUtilsSuite extends SparkFunSuite with SQLHelper {
     }
   }
 
+  test("SPARK-35680: format negative month intervals") {
+    import org.apache.spark.sql.types.YearMonthIntervalType._
```
[the remark this answers is not in the document]
Comment (nemanjapetr-db) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala:674]: Done.

**Comment by MaxGekk** [comment]
@nemanjapetr-db Please, add the `[TESTS]` tag to PR's title, see for instance https://github.com/apache/spark/pull/47620, and let's make PR's title shorter like `Add checks of negative months for toYearMonthIntervalString()`.

The changes LGTM.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaxGekk) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala:674]: I mean remove the import from other tests in the test suite, and put one line at the beginning of the file.

**Comment by MaxGekk** [comment]
The test failure seems not related to the changes:
```
[error] Error during tests:
[error] 	org.apache.spark.sql.jdbc.v2.OracleIntegrationSuite
```

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (uros-db) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala:674]: I also find it a bit weird to have imports in tests, however I see that this is actually common in `IntervalUtilsSuite.scala`

perhaps Nemanja can keep it as it is for now (and possibly follow up later with a fix, updating the entire suite)?

**Comment by nemanjapetr-db** [comment]
Included changes in HiveResultSuite.scala to this PR.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (nemanjapetr-db) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala]: Moved imports in both files. PTAL.

**Review by MaxGekk — COMMENTED** [review]
LGTM, waiting for CI.

**Comment by MaxGekk** [comment]
@nemanjapetr-db Could you remove this `…) in the context of negative months.` from PR's description.

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master.
Thank you, @nemanjapetr-db and @uros-db for review.

**Comment by MaxGekk** [comment]
@nemanjapetr-db Congratulations with your first contribution to Apache Spark!

**Pull request closed without merging**